### DISCUSSION TITLE
Part-DB inventory as KiCAD library?

### DISCUSSION THREAD

**Discussion by jbtronics** [Part-DB/Part-DB-server#395] [Polls]
Starting with version 7.0 [KiCAD](https://www.kicad.org/) offered the ability for remote libraries. The idea is that instead of having than to maintain local files with your commonly used components, you instead use a centralized database, which makes it especially useful for teams. As you often want to use components you already have instock, it seems useful to integrate it with an inventory managment system like Part-DB.

Until recently, KiCAD only offered remote libraries via direct database access, which is not really possible to realize with Part-DB. However, in the last weeks the experimental versions got support for remote libraries using HTTP, which is much easier to implement with Part-DB. This will most likely be part of the KiCAD 8.0 release, which will be released next year.

 The idea would be, that you could define some additional fields on your parts (and maybe categories and footprints), which says which symbol, footprint and reference string the part should use inside Part-DB. Then you can add your Part-DB instance as a KiCAD library and it will show you all available parts in your Part-DB database in KiCAD (with all info which are saved) and you can choose the one you need. Everybody, with access to the Part-DB instance, could directly see which parts are available for use, and you can configure each part centrally for your team.

I have done some experiments, and it should be fairly easy to implement the current API KiCAD requires into Part-DB. The main challenge would be to implement it in a user-friendly way. Currently, the KICAD API is read only, meaning that all definitions need to be done inside Part-DB somehow. Maybe there will some changes, which would allow to edit that from inside KiCAD too.

Now the question is, would you use such a feature / integration with KiCAD?

**Comment by Nudelsalad** [comment]
Hi there,
Glad to hear a maintainer is thinking of this feature. That would be fantastic.  But why exactly is it not possible to access the partdb database? because of the way the tables are distributed? I would love to see the feature where the datasheet can be referenced inside kicad and pulls directly from partdb!

**Reply by jbtronics** [reply]
On direct database access KiCAD wants all required data in one table as it seems it can not really do joins (which also would be pretty difficult to implement). Also some stuff like database pathes are not directly saved in the DB, but must be generated on the fly.
Also Part-DB is a webserver application, and normally you don't necessarily want to expose your application database to a network (especially not the internet).
Also this circumvents all of Part-DB User system and security measures.

By implementing the REST API for kicad. Part-DB can control who can access it (each user can generate their own read only access tokens), and give useful responses

**Reply by cooki371** [reply]
Is there already a desicion if this will be implemented?
If so: When is it planned to be released?
If there are no further plans yet: Is it possible/welcomed to participate implementing it?

**Comment by jbtronics** [comment]
This is now implemented in the master branch and will be part of Part-DB 1.10.0

See documentation: https://docs.part-db.de/usage/eda_integration.html

**Comment by iyzpx** [comment]
Fisrt time I installed a KiCad nightly to test this awesome feature and it aready works very well! Anyway, some questions for future using:

* Thats probably just KiCad, but do you know if there are any plans to edit fields in the future, or will it remain read only?
* ~There are now two footprint fields with different naming conventions. Edit each part would be a lot of work. Would it be an option to add the EDA footprint field to the footprint settings as well?~ Well, nvm. it's already there. But symbols needs to be edited anyways.
*  When I first read your plan for this, I thought that footprints and symbols will be stored in partdb. But that probably doesn't make much sense, does it?


![image](https://github.com/Part-DB/Part-DB-server/assets/103630942/a18dd836-28d5-4c0a-be20-0f488ad1bf15)

**Reply by jbtronics** [reply]
> Thats probably just KiCad, but do you know if there are any plans to edit fields in the future, or will it remain read only?

In the initial PR to KiCad it was mention as a future extension to the HTTP library API. I guess it would make the handling easier, if you could change the things from inside KiCad. But I guess thats a lot of work to implement in KiCad and will be not be part of KiCAD 8.

> Well, nvm. it's already there. But symbols needs to be edited anyways.

You can define symbols on a category level, and all parts with this category inherit this symbol (as long as not overwritten on the part itself). That makes sense for parts like resistors which are pretty much interchangable, but just have different values and footprints. For more complex parts like ICs you have to define it seperatly on each part. I dont really see a possibility to improve that really.

> When I first read your plan for this, I thought that footprints and symbols will be stored in partdb. But that probably doesn't make much sense, does it?

KiCad can only work with references to symbol and footprint files via that API. Part-DB can not really do anything about it. And I also think that it is useful just to store references. At least for symbols and footprints, which are already included in every KiCAD installtion.
Maybe it could be useful to store custom footprints/symbols on Part-DB for KiCad, so you dont have to distribute them to every KiCad user. But in the moment KiCad does not allow to transfer these custom symbols from the server to KiCad.

**Comment by ovidiu-leonte-nokto** [comment]
Hi! Is it possible to add the EDA Information fields to the list of parameters that can be displayed in the main window?
<img width="163" alt="Screenshot 2025-06-27 132013" src="https://github.com/user-attachments/assets/13929758-d40b-4f4e-846d-8c5cd7c77190" />

**Reply by mkne** [reply]
Do you think all EDA fields are necessary?
I agree that the following fields would be really useful: Value, KiCad footprint
Maybe also: KiCad schematic symbol
Not useful (from my current point of view): Force visibility, Exclude part from BOM, Exclude part from PCB/Board, Exclude part from simulation, Reference prefix

**Reply by ovidiu-leonte-nokto** [reply]
Yes, I agree with you.
Really useful would be Value and KiCad footprint. And if you could add these 2 fields to the Filter also it would be amazing!
The KiCad schematic symbol and Force visibility are nice to have to be able to see that the part is completed and has all the necessary fields for use in KiCad, because we have many parts that have been automatically created from info provider and don't have these fields completed.
The other fields I think are not useful to be displayed in the main window.
